### DISCUSSION TITLE
Brightway 2.5 support in Activity Browser? Give us feedback!

### DISCUSSION THREAD

**Discussion by marc-vdm** [LCA-ActivityBrowser/activity-browser#1174] [General]
We are currently exploring moving Activity Browser to Brightway 2.5.
This work is led by Chris Mutel and [Cauldron](https://www.cauldron.ch/). As this is a big project, this will take some time. When we know more concretely if/when this change will be complete we will share that.

What would a Brightway 2.5 version of Activity Browser mean for you?

- If you're only using Activity Browser (not Brightway):
  - Nothing major would change for you
  - There would be a wizard (the software kind, unfortunately 🧙‍♂️) that would guide you step-by-step to create Brightway 2.5 copies of your projects (and databases within those projects)
- If you're using Brightway 2 in the same projects as Activity Browser:
  - There would be a wizard that will guide you step-by-step to create Brightway 2.5 copies of your projects (and databases within those projects)
  - You would have to change your own scripts to Brightway 2.5 if you would want to keep using them in combination with Activity Browser
- If you're already using Brightway 2.5 in other projects than you use with Activity Browser:
  - Great news! Activity Browser would become compatible with those projects!
- Note that __Brightway 2.5 projects and databases are not backwards compatible with Brightway 2__

Right now, we are looking for community feedback on whether we should switch, and if we do, if you would want to keep support for a Brightway 2 version alongside the Brightway 2.5 version. The Brightway 2 version will keep existing either way, but we want to hear from you if you want a Brightway 2 version to keep receiving new features. An advantage of two versions is that we stay compatible for other libraries in the Brightway ecosystem that may not (yet) have switched to Brightway 2.5, ~~e.g. [LCA Algebraic](https://github.com/oie-mines-paristech/lca_algebraic/) (See the discussion over at lca_algebraic here: https://github.com/oie-mines-paristech/lca_algebraic/issues/42)~~ ([LCA Algebraic is moving to BW2.5](https://brightway.groups.io/g/development/topic/new_release_of/105686433)). A disadvantage would be that we are a small team with limited resources, developing two versions in parallel would give us a lot more work, while we could be spending that time on one version instead.

Let us know what you think, we want to hear from you on this!

**Comment by marc-vdm** [comment]
One comment we received over email:

>[...]
>
>And regarding the issue of keeping a bw2 version of AB, I think it would be a good option, as long as other brightway-LCA library are still only available for bw2. This is the case for lca-algebraic for instance, which is very helpful ! Keeping a version of AB on bw2 would give more time to the developer to update their library ?
>
>It is just a suggestion !
>
>Thank you for the amazing work !

**Comment by AdelineJerome** [comment]
Hi,

In my case, I started using the AB and then learned about how to use Brightway more recently. It means that all the training I got was on Bightway 2.5 and I also find the documentation more detailed for Brightway 2.5. However, I find it very useful to check my model and run calculations via the AB.
For now, I chose to try using Brightway 2, but I would very much appreciate it if the AB becomes compatible with Brightway 2.5. I could have access to my old projects in an environment with an older version of the AB.
I do not know how my case is representative of the rest of the community, but it might be the case for "new" Brightway users as well?

These are my two cents on the topic!

**Comment by aringot** [comment]
Hi,
I would like to use Brightway in one of my projects (I tried OpenLCA via olca-ipc, it was painful). I will likely need the Activity Browser at some point but it does not seem smart to use an outdated version of Brightway 🤔 

Wouldn't it make sense for Brightway to clearly encourage people to use the latest stable version (and use it in the Activity Browser as well)?

Is there any update on the status of this issue / is it still recommended to use Brightway 2? (I would prefer starting with 2.5 and could wait a few months for the AB to be available in Brightway 2.5)

Thank you!

**Reply by marc-vdm** [reply]
Thanks for sharing your thoughts!

> Wouldn't it make sense for Brightway to clearly encourage people to use the latest stable version (and use it in the Activity Browser as well)?

Yes! Which is why we're working on moving to Brightway 2.5, however, it's that takes quite a while due to the complexity how we communicate with Brightway. This is something that we're changing behind the screens as well, as part of our switch to Brightway 2.5, to make future changes easier.

>Is there any update on the status of this issue 

We are working on a beta version of Activity Browser working on Brightway 2.5, which will come out this year, but we cannot be more specific than this currently. 

> is it still recommended to use Brightway 2? (I would prefer starting with 2.5 and could wait a few months for the AB to be available in Brightway 2.5)

If you want to use both Brightway and activity browser right now, you HAVE to use Brightway 2, if you only use one, it doesn't matter. If you're willing to wait until later this year -again, we will not be more specific than that- I'd suggest waiting and starting with Brightway 2.5.


Hope that answers your questions

Marc

**Comment by marc-vdm** [comment]
Closing as outdated with the release of our Beta
https://brightway.groups.io/g/development/message/368
